### PR TITLE
AdvancedFilters: Create workable defaults for Reports that don't have them

### PR DESCRIPTION
**Pull request by psealock**
Fixes https://github.com/woocommerce/woocommerce-admin/issues/6529

Attempts to add advanced filters to reports that don't already have them was causing fatal errors when extension devs used the prescribed methods of extending.

This PR add better defaults and a check to see if any advanced filters have been added. If they have, add a regular filter "Advanced filters" so the newly added filters can be accessed.

### Screenshots

Screen Shot 2021-06-16 at 3 59 01 PM

### Detailed test instructions:
Instructions follow the first few steps of https://developer.woocommerce.com/2020/02/20/extending-wc-admin-reports in order to replicate Filters alongside Advanced Filters.

1. Create a new plugin if you don't already have one.

```
npm run create-wc-extension
cd ../<my-plugin-name>
npm install
npm start
```
2. Visit the Taxes Report and make sure filters work correctly.
3. Add an advanced filter in JS for the Taxes Report, which does not currently have any.

```js
const addShippedDateAdvancedFilter = ( advancedFilters ) => {
	advancedFilters.filters.Shipped = {
		labels: {
			add: __( 'Shipped Date', 'woocommerce-admin' ),
			remove: __( 'Remove Shipped Date filter', 'woocommerce-admin' ),
			rule: __( 'Select Shipped Date', 'woocommerce-admin' ),
			title: __(
				'{{title}}Shipped Date{{/title}} {{rule /}} {{filter /}}',
				'woocommerce-admin'
			),
			filter: __( 'Select Shipped Date', 'woocommerce-admin' ),
		},
		rules: [
			{
				value: 'between',
				label: __( 'Between', 'woocommerce-admin' ),
			},
		],
		input: {
			component: 'Date',
		},
	};

	return advancedFilters;
};

addFilter(
	'woocommerce_admin_taxes_report_advanced_filters',
	'lab-reporting-extension',
	addShippedDateAdvancedFilter
);
```
4. Visit the Taxes Report again and see the regular filters now have an option "Advanced filters"
5. Add an advanced filter and see the url update with parameters of the advanced filter

### CODE REVIEW

**Review by jeffstieler — COMMENTED** [review]
This code looks good and tests well, but were the Products and Revenue reports missed?

**Comment by psealock** [comment]
Products and Revenue added, this is ready for another look

File: client/analytics/report/categories/config.js [hunk 1]
```diff
@@ -44,73 +44,86 @@ export const charts = applyFilters( CATEGORY_REPORT_CHARTS_FILTER, [
...
-								),
-							},
-						},
-					},
-				],
-			},
+		filters: {},
+		title: _x(
+			'Categories Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/categories/config.js:53]: Variations should be Categories.

File: client/analytics/report/coupons/config.js [hunk 2]
```diff
@@ -35,68 +35,78 @@ export const charts = applyFilters( COUPON_REPORT_CHARTS_FILTER, [
...
-		label: __( 'Show', 'woocommerce-admin' ),
-		staticParams: [ 'chartType', 'paged', 'per_page' ],
-		param: 'filter',
-		showFilters: () => true,
-		filters: [
-			{ label: __( 'All Coupons', 'woocommerce-admin' ), value: 'all' },
+		filters: {},
+		title: _x(
+			'Coupons Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/coupons/config.js:44]: Variations should be Coupons.

File: client/analytics/report/products/config.js [hunk 3]
```diff
@@ -169,12 +169,30 @@ const variationsConfig = {
 	],
 };
 
+export const advancedFilters = applyFilters(
+	PRODUCTS_REPORT_ADVANCED_FILTERS_FILTER,
+	{
+		filters: {},
+		title: _x(
+			'Product Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/products/config.js:178]: Variations should be Products.

File: client/analytics/report/products/config.js [hunk 4]
```diff
@@ -169,12 +169,30 @@ const variationsConfig = {
 	],
 };
 
+export const advancedFilters = applyFilters(
+	PRODUCTS_REPORT_ADVANCED_FILTERS_FILTER,
+	{
+		filters: {},
+		title: _x(
+			'Product Match {{select /}} Filters',
```
Comment (jeffstieler) [inline client/analytics/report/products/config.js:177]: Should product be pluralized here?

File: client/analytics/report/revenue/config.js [hunk 5]
```diff
@@ -60,8 +60,32 @@ export const charts = applyFilters( REVENUE_REPORT_CHARTS_FILTER, [
...
+	} );
+	filterValues.push( {
+		label: __( 'Advanced Filters', 'woocommerce-admin' ),
+		value: 'advanced',
+	} );
+}
+
+export const filters = applyFilters(
+	REVENUE_REPORT_FILTERS_FILTER,
+	filterValues
```
Comment (jeffstieler) [inline client/analytics/report/revenue/config.js:90]: This isn't a complete config and results in the following console error: `An error occurred while running 'mapSelect': filterConfig.showFilters is not a function`.

File: client/analytics/report/revenue/config.js [hunk 6]
```diff
@@ -60,8 +60,32 @@ export const charts = applyFilters( REVENUE_REPORT_CHARTS_FILTER, [
...
 
-export const filters = applyFilters( REVENUE_REPORT_FILTERS_FILTER, [] );
 export const advancedFilters = applyFilters(
 	REVENUE_REPORT_ADVANCED_FILTERS_FILTER,
-	{}
+	{
+		filters: {},
+		title: _x(
+			'Revenue Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/revenue/config.js:69]: Variations should be Revenue.

File: client/analytics/report/revenue/config.js [hunk 7]
```diff
@@ -60,8 +60,32 @@ export const charts = applyFilters( REVENUE_REPORT_CHARTS_FILTER, [
...
 ] );
 
-export const filters = applyFilters( REVENUE_REPORT_FILTERS_FILTER, [] );
 export const advancedFilters = applyFilters(
 	REVENUE_REPORT_ADVANCED_FILTERS_FILTER,
-	{}
+	{
+		filters: {},
+		title: _x(
+			'Revenue Match {{select /}} Filters',
```
Comment (jeffstieler) [inline client/analytics/report/revenue/config.js:68]: Should this be "Revenue Matches"?

File: client/analytics/report/stock/config.js [hunk 8]
```diff
@@ -10,32 +10,48 @@ const STOCK_REPORT_ADVANCED_FILTERS_FILTER =
 
 export const showDatePicker = false;
 
+export const advancedFilters = applyFilters(
+	STOCK_REPORT_ADVANCED_FILTERS_FILTER,
+	{
+		filters: {},
+		title: _x(
+			'Taxes Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/stock/config.js:19]: Variations should be Products.

File: client/analytics/report/stock/config.js [hunk 9]
```diff
@@ -10,32 +10,48 @@ const STOCK_REPORT_ADVANCED_FILTERS_FILTER =
 
 export const showDatePicker = false;
 
+export const advancedFilters = applyFilters(
+	STOCK_REPORT_ADVANCED_FILTERS_FILTER,
+	{
+		filters: {},
+		title: _x(
+			'Taxes Match {{select /}} Filters',
```
Comment (jeffstieler) [inline client/analytics/report/stock/config.js:18]: Should this be Products?

File: client/analytics/report/taxes/config.js [hunk 10]
```diff
@@ -51,49 +51,65 @@ export const charts = applyFilters( TAXES_REPORT_CHARTS_FILTER, [
 	},
 ] );
 
+export const advancedFilters = applyFilters(
+	TAXES_REPORT_ADVANCED_FILTERS_FILTER,
+	{
+		filters: {},
+		title: _x(
+			'Taxes Match {{select /}} Filters',
+			'A sentence describing filters for Variations. See screen shot for context: https://cloudup.com/cSsUY9VeCVJ',
```
Comment (jeffstieler) [inline client/analytics/report/taxes/config.js:60]: Variations should be Taxes.

**Review by jeffstieler — CHANGES_REQUESTED** [review]
There's a bug with the Revenue report filter config, and some copy-pasta that needs cleaning up. 🍝

**Comment by psealock** [comment]
Note to self: This should be a better default

https://github.com/woocommerce/woocommerce-admin/blob/130eb95bae34a8efd83143c52cc08864efc15131/packages/components/src/filters/index.js#L225

**Comment by psealock** [comment]
Thanks for the review @jeffstieler.

I fixed the strings, addressed the Revenue Report problem you found and found another error in Stock Report. The filters modify a query param `type`, not `filter`, so if any advancedFilters have been added a new set of filters gets added.

This is ready for another look 🙇🏽‍♂️

**Review by jeffstieler — APPROVED** [review]
Code looks good and this is testing well. 🛳️   it!

**Comment by psealock** [comment]
Since this will be cherry-picked, changelog was added as part of https://github.com/woocommerce/woocommerce-admin/pull/7320

**Pull request merged**